app/src/main/scala/org/alephium/explorer/persistence/queries/InputQueries.scala

```diff
@@ -101,14 +99,14 @@ object InputQueries {
 
   // format: off
   def inputsFromTxsNoJoin(txHashes: Seq[Transaction.Hash]):
-    DBActionR[Seq[(Transaction.Hash, Int, Int, Hash, Option[String], Transaction.Hash, Address, U256)]] = {
+    DBActionR[Seq[(Transaction.Hash, Int, Int, Hash, Option[String], Address, U256)]] = {
   // format: on
     if (txHashes.nonEmpty) {
       val values = txHashes.map(hash => s"'\\x$hash'").mkString(",")
       sql"""
-    SELECT tx_hash, input_order, hint, output_ref_key, unlock_script, output_ref_tx_hash, output_ref_address, output_ref_amount
+    SELECT tx_hash, input_order, hint, output_ref_key, unlock_script, output_ref_address, output_ref_amount
     FROM inputs
-    WHERE tx_hash IN (#$values) AND main_chain = true AND output_ref_tx_hash IS NOT NULL
+    WHERE tx_hash IN (#$values) AND main_chain = true AND output_ref_address IS NOT NULL
```

Comment:
There are two issues with this query:
1. We will miss some inputs when `output_ref_address` is NULL, this will make the built transaction incomplete. (If I want to get all of the inputs&outputs of a TX, I don't care if the address is NULL).
2. `main_chain = true` can cause troubles too.

My suggestion:
1. use `Option[Address]` for `output_ref_address` so that the built transaction will always be complete
2. query inputs based on `(txHash, blockHash)` instead of `mainc_chain = true`. I think that this approach should be applied to all inputs&outputs queries when `(txHash, blockHash)` is available.

Reply:
1. Fully agree, front-end can just not show the address if it's missing, they could fall back to the output ref
2. I'll go back to this once I'm done merging master.

Reply:
2. Went back: Great idea, thinking out loud but it might even allows us to remove the `main_chain` in the inputs/outputs (I would need to check in the code tho).

Reply:
1. Done

moving to 2.

Reply:
2. Done

I also took the time to add the 3 fields we were discussing above, just in db for now, no need to add them to the api yet.

Reply:
we unfortunatelly won't be able to totally remove `main_chain`, we need it for example for the [getBalance query](https://github.com/alephium/explorer-backend/blob/61939e74e00eebed165ab43b3b030ced17f601fc/app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala#L375-L391)